tools/preview_new_release.sh

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env bash
+
+######################################################################
+#                                                                    #
+#                            OCamlFormat                             #
+#                                                                    #
+#  Copyright (c) 2017-present, Facebook, Inc.  All rights reserved.  #
+#                                                                    #
+#  This source code is licensed under the MIT license found in the   #
+#  LICENSE file in the root directory of this source tree.           #
+#                                                                    #
+######################################################################
+
+set -o errexit
+set -o nounset
+
+user=
+user_set=0
+version=
+version_set=0
+update_repos=0
+prefix=
+prefix_set=0
+
+function usage()
+{
+    echo "usage: $0 -user <GITHUB_USERNAME> -version <RELEASE> [--update-repositories] [--prefix <PREFIX>]"
+}
+
+while [[ $# -gt 0 ]] && [[ "$1" == "--"* ]]; do
+    opt="$1";
+    shift 1
+    case "$opt" in
+        "--user")
+            user="$1";
+            user_set=1;
+            shift 1;
+            ;;
+        "--version")
+            version="$1";
+            version_set=1;
+            shift 1;
+            ;;
+        "--update-repositories")
+            update_repos=1;
+            ;;
+        "--prefix")
+            prefix="$1";
+            prefix_set=1;
+            shift 1;
+            ;;
+        *)
+            usage;
+            exit 1;
+            ;;
+    esac
+done;
```

Comment:
There's a builtin function for parsing options but it can only work with single character options. It would look like that:
```suggestion
function usage()
{
    echo "usage: $0 -u <GITHUB_USERNAME> -v <RELEASE> [-U] [-p <PREFIX>]"
}

while getopts ":u:v:Up:" opt; do
    case "$opt" in
        u)
            user=$OPTARG
            user_set=1
            ;;
        v)
            version=$OPTARG
            version_set=1
            ;;
        U)
            update_repos=1;
            ;;
        p)
            prefix=$OPTARG
            prefix_set=1
            ;;
        *)
            usage;
            exit 1;
            ;;
    esac
done;
```